app/src/main/java/com/sopt/now/compose/screen/auth/signup/SignUpViewModel.kt

```diff
@@ -3,32 +3,26 @@ package com.sopt.now.compose.screen.auth.signup
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import com.sopt.now.ServicePool
+import androidx.lifecycle.viewModelScope
+import com.sopt.now.compose.repository.SignUpRepository
 import com.sopt.now.data.model.RequestSignUpDto
-import com.sopt.now.data.model.ResponseSignUpDto
-import retrofit2.Call
-import retrofit2.Callback
-import retrofit2.Response
+import kotlinx.coroutines.launch
 
-class SignUpViewModel : ViewModel() {
+class SignUpViewModel(
+    private val signUpRepository: SignUpRepository
+) : ViewModel() {
     private val _signUpState = MutableLiveData<SignUpState>()
     val signUpState: LiveData<SignUpState> = _signUpState
 
-    fun signUp(dto: RequestSignUpDto) {
-        _signUpState.value = SignUpState.Loading
-        val call = ServicePool.authService.signUp(dto)
-        call.enqueue(object : Callback<ResponseSignUpDto> {
-            override fun onResponse(call: Call<ResponseSignUpDto>, response: Response<ResponseSignUpDto>) {
-                if (response.isSuccessful) {
-                    _signUpState.value = SignUpState.Success(response.body()?.message ?: "회원가입 성공")
-                } else {
-                    _signUpState.value = SignUpState.Error(response.body()?.message ?: "회원가입 실패")
-                }
+    fun signUp(signUpDto: RequestSignUpDto) {
+        viewModelScope.launch {
+            _signUpState.value = SignUpState.Loading
+            val result = signUpRepository.signUp(signUpDto)
+            _signUpState.value = if (result.isSuccess) {
+                SignUpState.Success("회원가입 성공")
+            } else {
+                SignUpState.Error(result.exceptionOrNull()?.message ?: "회원가입 실패")
```

Comment:
```suggestion
            signUpRepository.signUp(
                signUpDto
            ).onSuccess {
                 _signUpState.value = SignUpState.Success("회원가입 성공")
            }.onFailure{
                SignUpState.Error(result.exceptionOrNull()?.message ?: "회원가입 실패")
            }
```
취향차이일 수는 있을 것 같은데 result를 따로 변수화하지 않고도 사용해줄 수 있을 것 같아요!